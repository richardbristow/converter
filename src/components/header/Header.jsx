import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import HeaderIconLink from './HeaderIconLink';
import Logo from '../shared/Logo';

const StyledHeader = styled.div`
  grid-column: 1 / -1;
  grid-row: 1 / 2;
  display: grid;
  grid-template-columns: ${props => (props.sidebarDocked ? '75px 1fr auto auto' : '75px auto 1fr auto auto')};
  align-items: center;
  padding-right: 10px;
`;

const Header = ({ sidebarDocked }) => (
  <StyledHeader sidebarDocked={sidebarDocked}>
    <div />
    {!sidebarDocked && <Logo sidebarDocked={sidebarDocked} />}
    <div />
    <HeaderIconLink type="Help" />
    <HeaderIconLink type="About" />
  </StyledHeader>
);

Header.propTypes = {
  sidebarDocked: PropTypes.bool.isRequired,
};

export default Header;
