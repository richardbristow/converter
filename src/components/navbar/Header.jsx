import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import icons from '../../constants/icons/icons';

import StyledLink from '../shared/StyledLink';
import Icon from '../shared/Icon';

const StyledHeader = styled.div`
  background-color: tomato;
  grid-area: 'header';
  grid-column: 1 / -1;
  display: grid;
  grid-template-columns: ${({ sidebarDocked }) => (sidebarDocked ? 'auto 1fr auto auto' : '75px auto 1fr auto auto')};
  align-items: center;
  padding-right: 10px;
`;

const StyledHamburger = styled.div`
  padding: 8px;
  margin: 0 auto;
`;

const StyledLogoLink = styled(Link)`
  text-decoration: none;
  color: black;
  &:visited {
    color: black;
  };
  h1 {
    margin: 0 0 0 20px;
    user-select: none;
  }
`;

const StyledHeaderIcon = styled.div`
  color: yellow;
  &:hover {
    color: purple;
  };
`;

const Header = ({ handleClick, sidebarDocked }) => (
  <StyledHeader sidebarDocked={sidebarDocked}>
    {!sidebarDocked && (
      <StyledHamburger onClick={handleClick}>
        <Icon icon={icons.hamburger} />
      </StyledHamburger>
    )}
    <StyledLogoLink to="/">
      <h1>Converter</h1>
    </StyledLogoLink>
    <div />
    <StyledLink header to="/help">
      <span>Help</span>
      <StyledHeaderIcon>
        <Icon size="20" icon={icons.help} />
      </StyledHeaderIcon>
    </StyledLink>
    <StyledLink header to="/about">
      <span>About</span>
      <StyledHeaderIcon>
        <Icon size="20" icon={icons.about} />
      </StyledHeaderIcon>
    </StyledLink>
  </StyledHeader>
);

Header.propTypes = {
  handleClick: PropTypes.func.isRequired,
  sidebarDocked: PropTypes.bool.isRequired,
};

export default Header;
