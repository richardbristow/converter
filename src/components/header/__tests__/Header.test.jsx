import React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';

const props = {
  handleClick: jest.fn(),
  sidebarDocked: false,
};

describe('Header', () => {
  it('should render the header', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should find 3 links (1 for the Logo, and 2 page links)', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper.find('HeaderIconLink')).toHaveLength(2);
    expect(wrapper.find('Styled(Link)')).toHaveLength(1);
  });
});
