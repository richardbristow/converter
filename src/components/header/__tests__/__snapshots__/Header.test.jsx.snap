// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the header 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    handleHamburgerClick={[Function]}
    sidebarDocked={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div />,
        <Logo
          sidebarDocked={false}
/>,
        <div />,
        <HeaderIconLink
          type="Help"
/>,
        <HeaderIconLink
          type="About"
/>,
      ],
      "sidebarDocked": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "sidebarDocked": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "type": "Help",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "type": "About",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div />,
          <Logo
            sidebarDocked={false}
/>,
          <div />,
          <HeaderIconLink
            type="Help"
/>,
          <HeaderIconLink
            type="About"
/>,
        ],
        "sidebarDocked": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "sidebarDocked": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": "Help",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "type": "About",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
